### DISCUSSION TITLE
Exception on startup

### DISCUSSION THREAD

**Discussion by Teramaks** [Auties00/WhatsappWeb4j#101] [General]
Version: 2.2.6

`Caused by: java.lang.ClassNotFoundException: jakarta.websocket.ContainerProvider
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[na:na]
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[na:na]
	... 29 common frames omitted`

Resolved adding this dependency [https://mvnrepository.com/artifact/org.cometd.java/cometd-java-client-websocket-javax/7.0.5](maven)

**Comment by Auties00** [comment]
> Version: 2.2.6
> 
> `Caused by: java.lang.ClassNotFoundException: jakarta.websocket.ContainerProvider at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[na:na] at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) ~[na:na] at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[na:na] ... 29 common frames omitted`
> 
> Resolved adding this dependency [https://mvnrepository.com/artifact/org.cometd.java/cometd-java-client-websocket-javax/7.0.5](maven)

Are you importing the library correctly? That dependency is for a websocket implementation, but tyrus is already provided. Are you running in a particular env?

**Reply by Teramaks** [reply]
> > Version: 2.2.6
> > `Caused by: java.lang.ClassNotFoundException: jakarta.websocket.ContainerProvider at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[na:na] at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) ~[na:na] at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[na:na] ... 29 common frames omitted`
> > Resolved adding this dependency [https://mvnrepository.com/artifact/org.cometd.java/cometd-java-client-websocket-javax/7.0.5](maven)
> 
> Are you importing the library correctly? That dependency is for a websocket implementation, but tyrus is already provided. Are you running in a particular env?

JDK 17, windows 11 21H2, runnig directly from IDE, no containerizing in docker, here is my pom.xml: [https://gist.github.com/Teramaks/0b3c927b26583909b4335b5fd176c482](url)

Instantination: [https://gist.github.com/Teramaks/c015e781c4ac15378ff27f9ec4aba4d6](url)

**Reply by Auties00** [reply]
> https://gist.github.com/Teramaks/0b3c927b26583909b4335b5fd176c482

That's weird. Can you test it without spring boot?

**Reply by Teramaks** [reply]
> > https://gist.github.com/Teramaks/0b3c927b26583909b4335b5fd176c482
> 
> That's weird. Can you test it without spring boot?

I've already tested. All working fine, getting qr code with no problems and listening orher events (mostly deleted messages event). Even tested in docker, no any problems found. Running with spring boot, not java core.

**Reply by Auties00** [reply]
> > > https://gist.github.com/Teramaks/0b3c927b26583909b4335b5fd176c482
> > 
> > 
> > That's weird. Can you test it without spring boot?
> 
> I've already tested. All working fine, getting qr code with no problems and listening orher events (mostly deleted messages event). Even tested in docker, no any problems found. Running with spring boot, not java core.

Yeah I mean without that dependency and without spring boot. Perhaps spring boot is the problem
